textsplitter/recursiveTextSplitter.go

```diff
@@ -68,13 +68,11 @@ func (r *RecursiveCharacterTextSplitter) SplitText(text string) []string {
 	finalChunks := []string{}
 	// Get appropriate separator to use
 	separator := r.separators[len(r.separators)-1]
-	for _, s := range r.separators {
-		if s == "" {
-			separator = s
-			break
-		}
-		if strings.Contains(text, s) {
-			separator = s
+	newSeparators := []string{}
+	for i, c := range r.separators {
+		if c == "" || strings.Contains(text, c) {
+			separator = c
+			newSeparators = r.separators[i+1:]
```

Comment:
Following the python implementation, here it should be:

```suggestion
        newSeparators := []string{}
	for i, s := range r.separators {
		if s == "" {
			separator = s
			break
		}

		if strings.Contains(text, s) {
			separator = s
			newSeparators = r.separators[i+1:]
```

Reply:
The code running logic should be the same, since it will always run this code block `if r.lengthFunction(s) < r.chunkSize {}` and the `newSeparators` will never be used when `separator` is empty string("").
But i'm okay with this change.